File: Backend/Environment/environment/environment/layout.vb
Imports System.Drawing.Text
Imports System.IO
Imports AeonLabs.Environment.Assembly.assemblyEnvironmentVarsClass

Public Class environmentLayoutClass

    Public Property menu As New assemblyMenuClass

    'MAIN LAYOUT design scheme
    Public Property MENU_BACK_COLOR As Color = Color.Black
    Public Property MENU_CLOSED_STATE As Integer = 40
    Public Property MENU_OPEN_STATE As Integer = 400

    'widgets & plugIns design scheme
    Public Property labelForeColor As Color
    Public Property linkLabelForeColor As Color
    Public Property buttonForecolor As Color
    Public Property editTextBackColor As Color
    Public Property controlWithSelectionBackcolor As Color
    Public Property buttonBackcolor As Color

    Public Property borderColor As Color

    Public Property PanelBackColor As Color
    Public Property PanelTransparencyIndex As Double

    ' Time interval to change background image on main form
    Public Property RandomBackgroundTimeInterval As New TimeSpan(0, 15, 0) '15 min timeout - integer (Hours, Minutes, Seconds) 
    'fonts
    Public fontText, fontTitle As New PrivateFontCollection()
    Public Property fontTitleFile As String = "TrajanPro.ttf"
    Public Property fontTextFile As String = "Roboto-Medium.ttf"

    'font text size
    Public Property menuTitleFontSize As Integer = 10
    Public Property subMenuTitleFontSize As Integer = 8
    Public Property buttonFontSize As Integer = 12
    Public Property SmallTextFontSize As Integer = 7
    Public Property RegularTextFontSize As Integer = 9
    Public Property DialogTitleFontSize As Integer = 12
    Public Property groupBoxTitleFontSize As Integer = 8
    'main color schemes
    Public Property buttonColor As Color = Color.DarkOrange
    Public Property dividerColor As Color = Color.FromArgb(253, 186, 49)
    Public Property colorMainMenu As Color = Color.FromArgb(35, 40, 45)
    Public Property colorMainPageHeader As Color = Color.FromArgb(253, 186, 49)

    Public Sub loadDefaults(envars As environmentVarsCore)
        'ToDo check default font files are present
        Dim FontFileName = New FileInfo(envars.fontsPath & fontTitleFile)
        FontFileName.Refresh()
        If FontFileName.Exists Then
            envars.layoutDesign.fontTitle.AddFontFile(envars.fontsPath & fontTitleFile)
        Else
            MessageBox.Show("font file not found. reinstall the program")
            Throw New Exception("font file not found")
        End If

        FontFileName = New FileInfo(envars.fontsPath & fontTextFile)
        FontFileName.Refresh()
        If FontFileName.Exists Then
            envars.layoutDesign.fontText.AddFontFile(envars.fontsPath & fontTextFile)
        Else
            MessageBox.Show("font file not found. reinstall the program")
            Throw New Exception("font file not found")
        End If
    End Sub
End Class

File: Release Candidate/Platform/Backend/Environment/environmentCore/layout.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Drawing.Text;
using System.IO;
using System.Runtime.CompilerServices;
using System.Windows.Forms;

namespace AeonLabs.Environment
{
    public class environmentLayoutClass
    {
        public environmentLayoutClass()
        {
            menu = new menuEnvironmentVarsClass.menuClass();
        }

        public Dictionary<string, string> externalImages = new Dictionary<string, string>();
        private menuEnvironmentVarsClass.menuClass _menu;

        public menuEnvironmentVarsClass.menuClass menu
        {
            [MethodImpl(MethodImplOptions.Synchronized)]
            get
            {
                return _menu;
            }

            [MethodImpl(MethodImplOptions.Synchronized)]
            set
            {
                _menu = value;
            }
        }

        // MAIN LAYOUT design scheme
        public Color MENU_BACK_COLOR { get; set; } = Color.Black;
        public int MENU_CLOSED_STATE { get; set; } = 40;
        public int MENU_OPEN_STATE { get; set; } = 400;
        public int PANEL_SCROOL_UNDERLAY { get; set; } = 30;

        // widgets & plugIns design scheme
        public Color labelForeColor { get; set; } = Color.White;
        public Color linkLabelForeColor { get; set; } = Color.White;
        public Color buttonForecolor { get; set; } = Color.White;
        public Color editTextBackColor { get; set; } = Color.White;
        public Color controlWithSelectionBackcolor { get; set; }
        public Color buttonBackcolor { get; set; }
        public Color borderColor { get; set; }
        public Color PanelBackColor { get; set; }
        public double PanelTransparencyIndex { get; set; }
        public int IconsDefaultSize { get; set; }

        // Time interval to change background image on main form
        public TimeSpan RandomBackgroundTimeInterval { get; set; } = new TimeSpan(0, 15, 0); // 15 min timeout - integer (Hours, Minutes, Seconds) 
                                                                                             // fonts
        public PrivateFontCollection fontText = new PrivateFontCollection(), fontTitle = new PrivateFontCollection();

        public string fontTitleFile { get; set; } = "TrajanPro.ttf";
        public string fontTextFile { get; set; } = "Roboto-Medium.ttf";

        // font text size
        public int menuTitleFontSize { get; set; } = 10;
        public int subMenuTitleFontSize { get; set; } = 8;
        public int buttonFontSize { get; set; } = 12;
        public int SmallTextFontSize { get; set; } = 7;
        public int RegularTextFontSize { get; set; } = 9;
        public int DialogTitleFontSize { get; set; } = 12;
        public int groupBoxTitleFontSize { get; set; } = 8;
        // main color schemes
        public Color buttonColor { get; set; } = Color.DarkOrange;
        public Color dividerColor { get; set; } = Color.FromArgb(253, 186, 49);
        public Color colorMainMenu { get; set; } = Color.FromArgb(35, 40, 45);
        public Color colorMainPageHeader { get; set; } = Color.FromArgb(253, 186, 49);

        public void loadDefaults(environmentVarsCore envars)
        {
            // ToDo check default font files are present
            var FontFileName = new FileInfo(envars.fontsPath + fontTitleFile);
            FontFileName.Refresh();
            if (FontFileName.Exists)
            {
                envars.layoutDesign.fontTitle.AddFontFile(envars.fontsPath + fontTitleFile);
            }
            else
            {
                MessageBox.Show("font file not found. reinstall the program");
                throw new Exception("font file not found");
            }

            FontFileName = new FileInfo(envars.fontsPath + fontTextFile);
            FontFileName.Refresh();
            if (FontFileName.Exists)
            {
                envars.layoutDesign.fontText.AddFontFile(envars.fontsPath + fontTextFile);
            }
            else
            {
                MessageBox.Show("font file not found. reinstall the program");
                throw new Exception("font file not found");
            }
        }
    }
}